### PR TITLE
[3681] Use the rename representation event handler to rename representations

### LINKED ISSUE #3681: Replace all rename event handlers by `RenameRepresentationEventHandler`

### PR DESCRIPTION
**Pull request by gcoutable**
# Pull request template

## General purpose
What is the main goal of this pull request?
- [ ] Bug fixes
- [ ] New features
- [ ] Documentation
- [ ] Cleanup
- [ ] Tests
- [ ] Build / releng

## Project management
- [ ] Has the pull request been added to the relevant project and milestone? (Only if you know that your work is part of a specific iteration such as the current one)
- [ ] Have the `priority:` and `pr:` labels been added to the pull request? (In case of doubt, start with the labels `priority: low` and `pr: to review later`)
- [ ] Have the relevant issues been added to the pull request?
- [ ] Have the relevant labels been added to the issues? (`area:`, `difficulty:`, `type:`)
- [ ] Have the relevant issues been added to the same project and milestone as the pull request?
- [ ] Has the `CHANGELOG.adoc` been updated to reference the relevant issues?
- [ ] Have the relevant API breaks been described in the `CHANGELOG.adoc`? (Including changes in the GraphQL API)
- [ ] In case of a change with a visual impact, are there any screenshots in the `CHANGELOG.adoc`? For example in `doc/screenshots/2022.5.0-my-new-feature.png`

## Architectural decision records (ADR)
- [ ] Does the title of the commit contributing the ADR start with `[doc]`?
- [ ] Are the ADRs mentioned in the relevant section of the  `CHANGELOG.adoc`?

## Dependencies
- [ ] Are the new / upgraded dependencies mentioned in the relevant section of the `CHANGELOG.adoc`?
- [ ] Are the new dependencies justified in the `CHANGELOG.adoc`?

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose
- [ ] Is the code properly tested? (Plain old JavaScript tests for business code and tests based on React Testing Library for the components)

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- [ ] Variables have a proper type
- [ ] Functions’ arguments have a proper type
- [ ] Functions’ return type are specified
- Hooks are properly typed:
	- [ ] `useMutation<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useQuery<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useSubscription<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useMachine<CONTEXT_TYPE, EVENTS_TYPE>(…)`
	- [ ] `useState<STATE_TYPE>(…)`
- [ ] All components have a proper typing for their props
- [ ] No useless optional chaining with `?.` (if the GraphQL API specifies that a field cannot be `null`, do not treat it has potentially `null` for example)
- [ ] Nullable values have a proper type (for example `let diagram: Diagram | null = null;`)

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose
- [ ] Are all the event handlers tested?
- [ ] Are the event processor tested?
- [ ] Is the business code (services) tested?
- [ ] Are diagram layout changes tested?

### Architecture
- [ ] Are data structure classes properly separated from behavioral classes?
- [ ] Are all the relevant fields final?
- [ ] Is any data structure mutable? If so, please write a comment indicating why
- [ ] Are behavioral classes either stateless or side effect free?

### How to test this PR?
_Please describe here the various use cases to test this pull request_

- [ ] Has the Kiwi TCMS test suite been updated with tests for this contribution?

### CODE REVIEW

File: packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/representation/services/RepresentationMetadataPersistenceService.java [hunk 1]
```diff
@@ -32,20 +34,35 @@ public class RepresentationMetadataPersistenceService implements IRepresentation
...
+        var optionalRepresentationId = new UUIDParser().parse(representationMetadata.id());
+
+        if (optionalRepresentationId.isPresent()) {
             var representationId = optionalRepresentationId.get();
-            this.representationMetadataCreationService.create(cause, representationId, projectId, representationMetadata.getLabel(), representationMetadata.getKind(),
-                    representationMetadata.getDescriptionId(), targetObjectId);
+
+            var exists = this.representationMetadataSearchService.existsById(representationId);
+            if (exists) {
+                this.representationMetadataUpdateService.updateRepresentationMetadataLabelById(cause, representationId, representationMetadata.label());
```
Comment (pcdavid) [inline packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/representation/services/RepresentationMetadataPersistenceService.java]: This seems correct (at the moment), but still conceptually strange to unconditionally interpret "overwrite an existing representationmetadata" as "update its label".

**Review by pcdavid — APPROVED** [review]

File: packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/representation/services/RepresentationMetadataPersistenceService.java (hunk 1 above)
Comment (gcoutable) [reply packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/representation/services/RepresentationMetadataPersistenceService.java]: Oh ! Indeed ! That should be an update of the whole metadata

File: packages/sirius-web/backend/sirius-web-domain/src/main/java/org/eclipse/sirius/web/domain/boundedcontexts/representationdata/RepresentationMetadata.java [hunk 2]
```diff
@@ -90,6 +91,17 @@ public Instant getLastModifiedOn() {
...
     }
 
+    public void updateLabel(ICause cause, String newLabel) {
+        if (!this.label.equals(newLabel)) {
+            this.label = newLabel;
+
+            var now = Instant.now();
+            this.lastModifiedOn = now;
+
+            this.registerEvent(new RepresentationMetadataUpdateEvent(UUID.randomUUID(), now, cause, this));
```
Comment (sbegaudeau) [inline packages/sirius-web/backend/sirius-web-domain/src/main/java/org/eclipse/sirius/web/domain/boundedcontexts/representationdata/RepresentationMetadata.java]: UpdateEvent -> UpdateEvent

File: packages/sirius-web/backend/sirius-web-domain/src/main/java/org/eclipse/sirius/web/domain/boundedcontexts/representationdata/events/RepresentationMetadataUpdateEvent.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import org.eclipse.sirius.web.domain.boundedcontexts.representationdata.RepresentationMetadata;
+
+import jakarta.validation.constraints.NotNull;
+
+/**
+ * Event fired when the metadata of a representation are updated.
+ *
+ * @author gcoutable
+ */
+public record RepresentationMetadataUpdateEvent(
```
Comment (sbegaudeau) [inline packages/sirius-web/backend/sirius-web-domain/src/main/java/org/eclipse/sirius/web/domain/boundedcontexts/representationdata/events/RepresentationMetadataUpdateEvent.java]: Updated

File: packages/sirius-web/backend/sirius-web-domain/src/main/java/org/eclipse/sirius/web/domain/boundedcontexts/representationdata/services/api/IRepresentationMetadataUpdateService.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+import org.eclipse.sirius.web.domain.services.IResult;
+
+/**
+ * Used to update existing representation metadata.
+ *
+ * @author gcoutable
+ */
+public interface IRepresentationMetadataUpdateService {
+
+    IResult<Void> updateRepresentationMetadataLabelById(ICause cause, UUID representationMetadataId, String label);
```
Comment (sbegaudeau) [inline packages/sirius-web/backend/sirius-web-domain/src/main/java/org/eclipse/sirius/web/domain/boundedcontexts/representationdata/services/api/IRepresentationMetadataUpdateService.java]: updateLabel is enough :)

File: packages/sirius-web/backend/sirius-web/src/test/java/org/eclipse/sirius/web/application/controllers/representations/RepresentationLifecycleControllerIntegrationTests.java [hunk 5]
```diff
@@ -149,7 +149,7 @@ public void givenRepresentationToRenameWhenMutationIsPerformedThenTheRepresentat
             assertThat(typename).isEqualTo(RenameRepresentationSuccessPayload.class.getSimpleName());
 
             assertThat(this.domainEventCollector.getDomainEvents()).hasSize(1);
-            assertThat(this.domainEventCollector.getDomainEvents()).anyMatch(RepresentationContentUpdatedEvent.class::isInstance);
+            assertThat(this.domainEventCollector.getDomainEvents()).anyMatch(RepresentationMetadataUpdateEvent.class::isInstance);
```
Comment (sbegaudeau) [inline packages/sirius-web/backend/sirius-web/src/test/java/org/eclipse/sirius/web/application/controllers/representations/RepresentationLifecycleControllerIntegrationTests.java]: We should be able to add back the check that I removed a looong time ago to ensure that it has been properly renamed

**Review by sbegaudeau — CHANGES_REQUESTED** [review]

**Review by sbegaudeau — APPROVED** [review]

**Comment by sbegaudeau** [comment]
Everything is now ok, it can be merged when the ECA check will be back online

**Pull request merged**